### PR TITLE
Set up Redux.

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Added redux, and required deps.
- Added fake machine listing for redux demo.
- Improved project structure.
- Added second `./run` for development.

QA:
- Open the demo link, there should be a couple of machines.

Fixes: https://github.com/canonical-web-and-design/MAAS-squad/issues/1237.

### CODE REVIEW

File: ui/src/app/App.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+// Filters will live inside their respective modules.
+const getMachines = (state, props) => state.machines;
+
+const getReadyMachines = createSelector(
+  [getMachines],
+  machines => machines.filter(machine => machine.ready)
+);
+
```
Comment (squidsoup) [inline ui/src/app/App.js]: Might be good to add another comment here like the previous one, such that action creators are in their own modules. Realise this is throwaway code...

File: ui/src/index.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
-import App from "./App";
+
+import "./scss/base.scss";
 import * as serviceWorker from "./serviceWorker";
+import App from "./app/App";
+import reducers from "./root-reducer";
+
+const store = createStore(
+  reducers,
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (squidsoup) [inline ui/src/index.js]: Maybe leave a comment here that once we're using redux middleware like sagas, we'll probably want to change this to use the [redux-devtools-extension](https://github.com/zalmoxisus/redux-devtools-extension#13-use-redux-devtools-extension-package-from-npm).

Comment (huwshimi) [reply ui/src/index.js]: Oh nice!

File: ui/src/app/App.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/App.js]: Done.

**Pull request merged**